test/index.test.js

```diff
@@ -0,0 +1,245 @@
+'use strict';
+
+const Bookshelf = require('bookshelf')(require('knex')({
+  client: 'sqlite3',
+  connection: {
+    filename: './test/test.sqlite3'
+  }
+}));
+const Hapi      = require('hapi');
+
+const Book  = Bookshelf.Model.extend({
+  tableName: 'books',
+  filter: function () {
+    return this;
+  }
+});
+const Genre = Bookshelf.Model.extend({ tableName: 'genres' });
+
+const handler = (request, reply) => {
+  reply({});
+};
+
+describe('plugin', () => {
+
+  let server;
+
+  beforeEach(() => {
+    server = new Hapi.Server();
+    server.connection({ port: 80 });
+    server.register([
+      require('inject-then'),
+      {
+        register: require('hapi-query-filter'),
+        options: { ignoredKeys: ['include'] }
+      },
+      require('../lib')
+    ], () => {});
+  });
+
+  it('appends the total for models that have a filter function', () => {
+    server.route({
+      method: 'GET',
+      path: '/books',
+      config: {
+        plugins: {
+          totalCount: { model: Book }
+        },
+        handler
+      }
+    });
+
+    return server.injectThen({
+      method: 'GET',
+      url: '/books?include[]=total_count',
+      credentials: {}
+    })
+    .then((res) => {
+      expect(res.result.total_count).to.eql(0);
```

Comment:
it seems a bit weird that we're just asserting that the counts equal 0 for these tests. could we actually add rows (and potentially a filter function that reduces that count) to make sure it's doing what it's supposed to?